packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -573,18 +577,21 @@ export class GarbageCollector implements IGarbageCollector {
 
 		// 4. Run the Sweep phase.
 		// It will delete sweep ready nodes and return a list of deleted node ids.
-		const deletedNodeIds = this.runSweepPhase(
-			gcResult,
-			sweepReadyNodeIds,
-			currentReferenceTimestampMs,
-			logger,
-		);
+		const deletedNodeIds = this.runSweepPhase(gcResult, sweepReadyNodeIds);
 
 		this.gcDataFromLastRun = cloneGCData(
 			gcData,
 			(id: string) => deletedNodeIds.includes(id) /* filter out deleted nodes */,
 		);
-		return gcStats;
+
+		// 5. Get the sweep phase stats.
+		const sweepPhaseStats = this.getSweepPhaseStats(
+			this.configs.shouldRunSweep ? this.deletedNodes : sweepReadyNodeIds,
```

Comment:
It's rare but `configs.shouldRunSweep` could be overridden to false but there have been nodes deleted before.

So I think this should be just a merge of `this.deletedNodes` and `sweepReadyNodeIds`. What do you think?

Reply:
or... shouldn't `deletedNodeIds` factor in here, maybe instead of `sweepReadyNodeIds`? In case ContainerRuntime doesn't actually delete them all?

Reply:
I think what you want to pass in is `Nodes already deleted before this GC run` and `Nodes deleted during this GC run`.  So maybe we stash `this.deletedNodes` before running Sweep for the first one and pass `deletedNodes` for the second?

Reply:
Hmmm, that scenario does pose a problem. However, this will become simpler once the delete op change comes in because then we won't actually delete anything during sweep. Then we will have "previously deleted nodes" + "nodes deleted in this run".

Reply:
So, maybe we check this in now as-is and update once delete op change is in.

Reply:
AB#6403 to track updating the logic